recipes/b2/standard/conanfile.py

```diff
@@ -77,9 +77,8 @@ def build(self):
         os.chdir(build_dir)
         command = os.path.join(
             engine_dir, "b2.exe" if use_windows_commands else "b2")
-        full_command = \
-            "{0} --ignore-site-config --prefix=../output --abbreviate-paths install".format(
-                command)
+        full_command = "{0} --ignore-site-config --prefix=../output --abbreviate-paths" \
+                       " toolset={1} install".format(command, self.options.toolset)
```

Comment:
Will this work in case of `self.options.toolset == 'auto'`?

Reply:
Several lines above we have an if statement for such a case.

Reply:
It works for sure. the real command of `full_command`, `b2` here, supports `toolset=auto`, which is equivalent to not passing `toolset`. Furthermore, the `if` statement you mentioned is mandatory, but these two `toolset` arguments aren't the same case. At the above one, the real command is `build.sh`, which doesn't accept `auto` as a `toolset` name. On the contrary, it would like to guess the `toolset` if we pass nothing to it.
As a result, for the first one, we do use `if` statement because we have to. But for the second one, we don't need. But yes, we can also use `if` to filter `auto` here. I just don't think it's elegant :)

Reply:
Great explanation, @pichi-router.